### PR TITLE
Feature/refactor pd controller

### PR DESCRIPTION
**Pull request by rjgriffin42**
The differences between the different implementations of the same PD controller have always bothered me, so I fixed it here.

### CODE REVIEW

File: ihmc-robotics-toolkit/src/main/java/us/ihmc/robotics/controllers/AbstractPDController.java [hunk 1]
```diff
@@ -46,49 +63,24 @@ public double getRateError()
...
-      actionD.set(getDerivativeGain() * rateError.getDoubleValue());
+      actionP.set(proportionalGain.getValue() * positionError.getDoubleValue());
+      actionD.set(derivativeGain.getValue() * rateError.getDoubleValue());
 
       return actionP.getDoubleValue() + actionD.getDoubleValue();
    }
 
    public double computeForAngles(double currentPosition, double desiredPosition, double currentRate, double desiredRate)
    {
       //      System.out.println("PGain: " + proportionalGain.getDoubleValue() + "DGain: " + derivativeGain.getDoubleValue());
```
Comment (james-p-foster) [inline ihmc-robotics-toolkit/src/main/java/us/ihmc/robotics/controllers/AbstractPDController.java:77]: If we're yo-variablizing here, I'd ditch this comment. Or LogTools it

File: ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/referenceFrames/ReferenceFramesVisualizer.java
Comment (james-p-foster) [inline ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/referenceFrames/ReferenceFramesVisualizer.java:1]: Does this belong in this PR? (I'd like to see this particular change merged ASAP)

**Review by james-p-foster — APPROVED** [review]
One minor Q, one nit

File: ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/referenceFrames/ReferenceFramesVisualizer.java
Comment (rjgriffin42) [reply ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/referenceFrames/ReferenceFramesVisualizer.java:1]: No, this snuck in, but I'm fine with merging it regardless

**Review by Beomyeong — APPROVED** [review]

**Pull request merged**